api/v1beta1/azuremachine_validation.go

```diff
@@ -378,3 +392,45 @@ func ValidateDiagnostics(diagnostics *Diagnostics, fieldPath *field.Path) field.
 
 	return allErrs
 }
+
+// ValidateConfidentialCompute validates the configuration options when the machine is a Confidential VM.
```

Comment:
Do you think it would be worth adding a quick reference links for these validations? I was referring to the below ones, but still had some challenges in understanding the validations. 

- https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update?tabs=HTTP#vmdisksecurityprofile
- https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update?tabs=HTTP#securitytypes
- https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update?tabs=HTTP#uefisettings

Reply:
// TODO(@nawazkh) review these validations again for sanity check.

Reply:
@nawazkh thank you for the sanity check! As mentioned in [this](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3265#discussion_r1188415831) comment, I tried to add comments on the validations. But I'm not quite sure how helpful they are. I absolutely agree though that having the docs linked is quite helpful. Please let me know if you have more suggestions on how to make it more readable and clearer.